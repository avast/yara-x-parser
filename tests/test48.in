//Rule comment
rule test
{
	//Rule block comment
	strings:
		$a = "foo"
		$b = "bar"
	condition:
		any of ($a*, $b)
}
