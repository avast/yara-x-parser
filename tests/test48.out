SOURCE_FILE@0..118
  RULE@0..117
    COMMENT@0..14 "//Rule comment"
    WHITESPACE@14..15 "\n"
    RULE_KW@15..19 "rule"
    WHITESPACE@19..20 " "
    IDENTIFIER@20..24 "test"
    WHITESPACE@24..25 "\n"
    BLOCK_EXPR@25..117
      L_BRACE@25..26 "{"
      WHITESPACE@26..28 "\n\t"
      STRINGS@28..84
        COMMENT@28..48 "//Rule block comment"
        WHITESPACE@48..50 "\n\t"
        STRINGS_KW@50..57 "strings"
        COLON@57..58 ":"
        WHITESPACE@58..61 "\n\t\t"
        VARIABLE_STMT@61..71
          VARIABLE@61..63 "$a"
          WHITESPACE@63..64 " "
          ASSIGN@64..65 "="
          WHITESPACE@65..66 " "
          PATTERN@66..71
            STRING_LIT@66..71 "\"foo\""
        WHITESPACE@71..74 "\n\t\t"
        VARIABLE_STMT@74..84
          VARIABLE@74..76 "$b"
          WHITESPACE@76..77 " "
          ASSIGN@77..78 "="
          WHITESPACE@78..79 " "
          PATTERN@79..84
            STRING_LIT@79..84 "\"bar\""
      WHITESPACE@84..86 "\n\t"
      CONDITION@86..115
        CONDITION_KW@86..95 "condition"
        COLON@95..96 ":"
        WHITESPACE@96..99 "\n\t\t"
        BOOLEAN_TERM@99..115
          OF_EXPR@99..115
            QUANTIFIER@99..102
              ANY_KW@99..102 "any"
            WHITESPACE@102..103 " "
            OF_KW@103..105 "of"
            WHITESPACE@105..106 " "
            PATTERN_IDENT_TUPLE@106..115
              L_PAREN@106..107 "("
              VARIABLE_WILDCARD@107..110
                VARIABLE@107..109 "$a"
                STAR@109..110 "*"
              COMMA@110..111 ","
              WHITESPACE@111..112 " "
              VARIABLE_WILDCARD@112..114
                VARIABLE@112..114 "$b"
              R_PAREN@114..115 ")"
      WHITESPACE@115..116 "\n"
      R_BRACE@116..117 "}"
  WHITESPACE@117..118 "\n"
